ZigBeeValve/zigbee_test.go

```diff
@@ -0,0 +1,363 @@
+package main
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"io"
+	"log"
+	"net/http"
+	"strings"
+	"testing"
+
+	"github.com/sdoque/mbaigo/components"
+	"github.com/sdoque/mbaigo/forms"
+	"github.com/sdoque/mbaigo/usecases"
+)
+
+// mockTransport is used for replacing the default network Transport (used by
+// http.DefaultClient) and it will intercept network requests.
+
+type mockTransport struct {
+	returnError bool
+	resp        *http.Response
+	hits        map[string]int
+	err         error
+}
+
+func newMockTransport(resp *http.Response, retErr bool, err error) mockTransport {
+	t := mockTransport{
+		returnError: retErr,
+		resp:        resp,
+		hits:        make(map[string]int),
+		err:         err,
+	}
+	// Highjack the default http client so no actuall http requests are sent over the network
+	http.DefaultClient.Transport = t
+	return t
+}
+
+// TODO: this might need to be expanded to a full JSON array?
+
+const discoverExample string = `[{
+		"Id": "123",
+		"Internalipaddress": "localhost",
+		"Macaddress": "test",
+		"Internalport": 8080,
+		"Name": "My gateway",
+		"Publicipaddress": "test"
+	}]`
+
+// RoundTrip method is required to fulfil the RoundTripper interface (as required by the DefaultClient).
+// It prevents the request from being sent over the network and count how many times
+// a domain was requested.
+
+var errHTTP error = fmt.Errorf("bad http request")
+
+func (t mockTransport) RoundTrip(req *http.Request) (resp *http.Response, err error) {
+	t.hits[req.URL.Hostname()] += 1
+	if t.err != nil {
+		return nil, t.err
+	}
+	if t.returnError != false {
+		req.GetBody = func() (io.ReadCloser, error) {
+			return nil, errHTTP
+		}
+	}
+	t.resp.Request = req
+	return t.resp, nil
+}
+
+////////////////////////////////////////////////////////////////////////////////
+
+func TestUnitAsset(t *testing.T) {
+	// Create a form
+	f := forms.SignalA_v1a{
+		Value: 27.0,
+	}
+
+	// Creates a single UnitAsset and assert it changes
+	ua := initTemplate().(*UnitAsset)
+
+	// Change Setpt
+	ua.setSetPoint(f)
+
```

Comment:
Nitpick, remove this kind of empty lines (as you've done in the next test function).